src/adam_core/propagator/propagator.py

```diff
@@ -575,6 +577,29 @@ def propagate_orbits(
         if propagated_variants is not None:
             propagated = propagated_variants.collapse(propagated)
 
-        return propagated.sort_by(
+        # Return the results with the original origin and frame
+        # Preserve the original output origin for the input orbits
+        # by orbit id
+        final_results = None
+        unique_origins = pc.unique(orbits.coordinates.origin.code)
+        for origin_code in unique_origins:
+            origin_orbits = orbits.select("coordinates.origin.code", origin_code)
+            result_origin_orbits = propagated.where(
+                pc.field("orbit_id").isin(origin_orbits.orbit_id)
+            )
```

Comment:
Not a blocker, is this comparable in speed?
```
result_origin_orbits = propagated.apply_mask(pc.is_in(propagated.orbit_id, origin_orbits.orbit_id))
```

Reply:
I hadn't even considered these would be different. Have you seen differences between using `pc.field` and passing the column directly?

Reply:
It does look like your version is slightly faster. I filtered 100k orbits against 20k orbit ids. 
```
Existing: 0.00540
New:      0.00497
```

Pushing an update now.